react/Accordion/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import Icon from '../Icon'
+import styles from './styles.styl'
+
+class AccordionItem extends Component {
+  constructor(props) {
+    super(props)
+    this.onClick = this.onClick.bind(this)
+    this.state = { selected: false }
+  }
+
+  render() {
+    const { children, label, className } = this.props
+
+    const { selected } = this.state
+
+    const bodyName = label.replace(/\s+/g, '')
+
+    return (
+      <div className={cx(styles['c-accordion-item'], className)}>
+        <div
+          className={cx(
+            styles['c-accordion-title'],
+            {
+              [styles['is-active']]: selected
+            },
+            className
+          )}
+          aria-selected={selected}
+          aria-controls={bodyName}
+          role="tab"
+          tabIndex="0"
+          onClick={this.onClick}
+        >
+          <Icon icon="forward" />
+          {label}
+        </div>
+        <div
+          className={cx(
+            styles['c-accordion-body'],
+            {
+              [styles['is-active']]: selected
+            },
+            className
+          )}
+          aria-hidden={!selected}
+          aria-labelledby={bodyName}
+          role="tabpanel"
+        >
+          {children}
+        </div>
+      </div>
+    )
+  }
+
+  onClick() {
+    this.setState(previousState => ({ selected: !previousState.selected }))
+  }
+}
+
+class Accordion extends Component {
+  render() {
+    const { children, className } = this.props
+    return (
+      <div className={cx(styles['c-accordion'], className)}>{children}</div>
+    )
+  }
+}
+
+Accordion.propTypes = {
+  className: PropTypes.string
+}
+
+AccordionItem.propTypes = {
+  label: PropTypes.string.isRequired,
+  className: PropTypes.string
+}
+
+Object.assign(Accordion, { AccordionItem })
+
+export default Accordion
```

Comment:
A small tip: you can use `...` operator to deconstruct/construct an object:

```js
export default {
  ...Accordion,
  AccordionItem
}
```

Take a look at [how babel handles this syntax with `Object.assign`](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAECCzAPYCcAmBLRA7aBvAUAL776iQzxJqZYCSALgKYC2eRJjAHgA4r3SpGAMzABXEPwLRoAOjmUUGbABp80hdWwMWRIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=).

Reply:
Well I tried but not sure Styleguidist really gets it.